src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,294 @@
+                    .WithAdditionalAnnotations(WarningAnnotation),
+                    externs: default, 
+                    usings: default,
+                    members: compilationUnit.Members);
+                return compilationUnit.WithMembers(new SyntaxList<MemberDeclarationSyntax>(targetNamespaceDecl))
+                    .WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // We should have a single member which is a namespace declaration in this compilation unit.
+            var namespaceDeclaration = root.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+
+            // Move everything to global namespace
+            if (targetNamespaceParts.Length == 1 && targetNamespaceParts[0].Length == 0)
+            {
+                var triviaFromNamespaceDecl = GetOpeningAndClosingTriviaOfNamespaceDeclaration(namespaceDeclaration);
+
+                var members = namespaceDeclaration.Members;
+                var eofToken = compilationUnit.EndOfFileToken
+                    .WithAdditionalAnnotations(WarningAnnotation);
+
+                // Try to preserve trivia from original namesapce declaration.
+                // If there's any member inside the declaration, we attach them to the 
+                // first and last member, otherwise, simply attach all to the EOF token.
+                if (members.Count > 0)
+                {
+                    var first = members.First();
+                    var firstWithTrivia = first.WithPrependedLeadingTrivia(triviaFromNamespaceDecl.openingTrivia);
+                    members = members.Replace(first, firstWithTrivia);
+
+                    var last = members.Last();
+                    var lastWithTrivia = last.WithAppendedTrailingTrivia(triviaFromNamespaceDecl.closingTrivia);
+                    members = members.Replace(last, lastWithTrivia);
+                }
+                else
+                {
+                    eofToken = eofToken.WithPrependedLeadingTrivia(
+                        triviaFromNamespaceDecl.openingTrivia.Concat(triviaFromNamespaceDecl.closingTrivia));
+                }
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
+                    compilationUnit.AttributeLists, 
+                    members,
+                    eofToken).WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // Change namespace name
+            return root.ReplaceNode(namespaceDeclaration, 
+                namespaceDeclaration.WithName(
+                    CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithTriviaFrom(namespaceDeclaration.Name)
+                    .WithAdditionalAnnotations(WarningAnnotation, Formatter.Annotation)));
```

Comment:
Nit: i find reading code of the following form very hard to reason about:

```c#
if (condition)
{
    // lots and lots of code.
    return;
}

// simple stuff
return stuff;
```

The reason here is that it's difficult to understand what this code now applies to.  When the blocks are long on both ends, that often can't be helped.  But it's often clearer to change the above to:

```c#
if (!condition)
{
    // simpleStuff
    return;
}

// lots and lots of stuff
return stuff;
```

Reply:
This is because both 'branches' of the 'if' in this case are easy to associate back with the condition.  unlike in the former case where the second branch is far removed and you often have to go backtrack to recall what it was.

Note: this is a nit.  you don't need to cahnge anything here if you prefer this way :)